juniper_codegen/src/lib.rs

```diff
@@ -549,6 +550,114 @@ pub fn graphql_scalar(attr: TokenStream, body: TokenStream) -> TokenStream {
         .into()
 }
 
+/// `#[derive(GraphQLScalarValue)]` macro for deriving a [`ScalarValue`]
+/// implementation.
+///
+/// To derive [`ScalarValue`] on enum you should mark corresponding enum
+/// variants with `as_int`/`as_float`/`as_string`/`into_string`/`as_str`/
+/// `as_boolean` attributes (names correspond to [`ScalarValue`] required
```

Comment:
Let's rename `as_bool` directly in the trait, so it complies with Rust types naming. And remove `as_boolean` variant totally.